include/kernel.h

```diff
@@ -4180,6 +4180,7 @@ static inline char *K_THREAD_STACK_BUFFER(k_thread_stack_t *sym)
 
 #endif /* _ARCH_DECLARE_STACK */
 
+#if defined(CONFIG_USERSPACE)
```

Comment:
the doxygen group "mem_domain_apis" will not be defined if CONFIG_USERSPACE is not defined, and there are references to this doxygen group in doc/api/kernel_api.rst  that will throw an error.

Reply:
IIRC, the way we do this with other conditionally compiled subsystems is to not ifdef the API definitions in the headers, but instead just conditionally compile the implementation, so that the user gets errors at link time if they try to use something that isn't enabled.

Reply:
I understand the issue now, I was wondering what was causing this.
@dbkinder why are we doing preprocessing when generating doc using doxygen? is there a specific reason? incidentally we are missing out on a lot of information since a lot the code would depend on a config. Can we do, "ENABLE_PREPROCESSING = y" for the doxygen?. I think that was the control for doxygen, AFAIK.

Reply:
@andrewboie I think there might be some issue regarding that solution. I want to hide the k_mem_domain and k_mem_partition under CONFIG_USERSPACE. But the mem_domain api needs these structures to even compile. Thus causing a compilation issue and not a linker issue. I will see if there is a better way.

Reply:
what is the point of hiding these structs?
just get rid of the ifdef!

Reply:
ENABLE_PREPROCESSING=YES is the doxygen setting we're using (see doc/zephyr.doxyfile)

Reply:
ENABLE_PREPROCESSING has been enabled since I've been with the project... Not sure of the impact of removing it but I can check and see how it would impact the documentation.

Reply:
enabling preprocessing will not help you because CONFIG_USERSPACE is not on by default

@AdithyaBaglody please get rid of the ifdef. We don't care about exposing the structs. We don't conditionally compile any of the k_poll API definitions even though the implementation is selectively compiled with K_POLL